datadog-sidecar/src/service/sidecar_server.rs

```diff
@@ -422,6 +535,22 @@ impl SidecarServer {
     }
 }
 
+pub fn path_for_telemetry(instance_id: &InstanceId) -> CString {
+    let mut hasher = ZwoHasher::default();
+    instance_id.runtime_id.hash(&mut hasher);
```

Comment:
runtime_id makes little sense - after all you want to share to many processes.
A specific runtime id may process many different service+env+version tuples over time.

What we want is service, env and version here.